### PR TITLE
Improved type hints and API documentation

### PR DESCRIPTION
**Pull request by jeff-cohere**
This PR adds type hints and API documentation for the DTS client, also fixing some style issues.

### CODE REVIEW

**Comment by jeff-cohere** [comment]
I know Sphinx is the favorite tool for generating Python API documentation, but it's also hot garbage! :-)

File: dts/transfer_status.py [hunk 1]
```diff
@@ -3,7 +3,12 @@
 
 @dataclass(slots = True)
 class TransferStatus(object):
-    """`TransferStatus` - holds status information for a file transfer"""
+    """`TransferStatus` status information for a file transfer.
```
Comment (ialarmedalien) [inline dts/transfer_status.py:6]: no need for `(object)` again

File: dts/client.py [hunk 2]
```diff
@@ -11,13 +11,14 @@
...
 )
+from typing import Any
 
 logger = logging.getLogger('dts')
 api_version = 1
 
 class KBaseAuth(AuthBase):
     """Attaches a KBase-sensible Authorization header to the given Request object."""
-    def __init__(self, api_key):
+    def __init__(self, api_key: str):
```
Comment (ialarmedalien) [inline dts/client.py:21]: `self` can be given the type `"KBaseAuth"` and the output of the function is `None`, i.e.

```py
def __init__(self: "KBaseAuth", api_key: str) -> None:
```

Comment (ialarmedalien) [reply dts/client.py:21]: (self can be assigned the type `"KBaseAuth"` throughout the class)

File: dts/database.py [hunk 3]
```diff
@@ -2,7 +2,12 @@
 
 @dataclass(slots = True)
 class Database(object):
-    """`Database` - A database storing files that can be selected and transferred"""
+    """`Database` - A database storing files that can be selected and transferred.
```
[the remark this answers is not in the document]
Comment (jeff-cohere) [reply dts/database.py:5]: Thanks. I think this code predates your correcting me earlier. :-)

File: dts/client.py (hunk 2 above)
Comment (jeff-cohere) [reply dts/client.py:21]: cheesy, but good to know!

**Comment by jeff-cohere** [comment]
Thanks for the review, @ialarmedalien ! Looks like the CFS is down at the moment, so the service isn't running, but when it's back up again I'll push changes.

**Review by ialarmedalien — APPROVED** [review]

**Pull request merged**